pennylane/ops/qubit/observables.py

```diff
@@ -247,6 +250,14 @@ def _eigvals(cls, *params):
         w[idx] = 1
         return w
 
+    @classmethod
+    def _matrix(cls, *params):
+        basis_state = params[0]
+        m = np.zeros((2 ** len(basis_state), 2 ** len(basis_state)))
+        idx = int("".join(str(i) for i in basis_state), 2)
+        m[idx, idx] = 1
+        return m
```

Comment:
@mariaschuld I copied this from your other PR :)

Reply:
I guess it would be _really_ good to make the codebase avoid using `_matrix` and instead use a sparse version :sweat_smile:

Reply:
Yes, coming up soon :)